aten/src/ATen/templates/ExtensionBackendRegistration.h

```diff
@@ -4,11 +4,23 @@
 
 namespace at {
 
+template <typename FnPtr>
+inline std::set<std::string>& get_valid_schemas() {
+  static std::set<std::string> schemas;
+  return schemas;
+}
+
+${get_valid_schemas_specializations}
+
 template <typename FnPtr>
 inline void register_extension_backend_op(
     Backend backend,
     const char * schema,
     FnPtr fn) {
+      auto & valid_schemas = get_valid_schemas<FnPtr>();
+      if (valid_schemas.find(schema) == valid_schemas.end()) {
+        AT_ERROR("Attempted to register invalid combination of function and schema for: ", schema);
```

Comment:
this error message doesn't seem great, because you don't know as a user if you provided a totally bogus schema string or just your function pointer type is wrong.

Ideally I think this would work as follows:
1) if you provided a function name that doesn't match a known function, you get an error message that that function name doesn't exist.
2) if you provide a schema with a valid name but invalid schema, you get an error message that the schema doesn't match and print all the valid schemas for a function with that name.
3) (bonus points) if you provide a valid schema but an invalid function pointer type, you get some reasonable error message.  We have the C++ function string (not schema) we generate corresponding to each schema which we can print.  Ideally we could generate a string corresponding to the function pointer, but that might be more work than its worth now.

Reply:
okay, that seems reasonable and I went ahead and did all 3